kong/plugins/datadog/statsd_logger.lua

```diff
@@ -19,18 +19,33 @@ local stat_types = {
 local statsd_mt = {}
 statsd_mt.__index = statsd_mt
 
+local env_datadog_agent_host = os.getenv 'KONG_DATADOG_AGENT_HOST'
+local env_datadog_agent_port = os.getenv 'KONG_DATADOG_AGENT_PORT'
 
 function statsd_mt:new(conf)
   local sock   = udp()
-  local _, err = sock:setpeername(conf.host, conf.port)
+
+  local host, port
+  if conf.use_env then
+    if not (env_datadog_agent_host and env_datadog_agent_port) then
+      return nil, "both KONG_DATADOG_AGENT_HOST and KONG_DATADOG_AGENT_PORT must be set"
+    end
+    host = env_datadog_agent_host
+    port = tonumber(env_datadog_agent_port)
```

Comment:
```suggestion
    port = env_datadog_agent_port
```
if you convert here, then if the port was a string `"xxx"` for example, then by now it would be `nil`, and fail later on.

Reply:
done and done, thanks. Moving on the tests